syntax = "proto3";

import "role.proto";


message Agent {

	// The agent/participant's unique public key
	string public_key = 1;

	// A human-readable name identifying the agent
	string name = 2;

	// Approximately when the agent was registered, as a Unix UTC timestamp
	uint64 timestamp = 3;

	string description = 4;

	Role role = 5;
	repeated string orders = 6;
	
}

message AgentContainer {
	repeated Agent entries = 1;
}